### PR TITLE
Submodule conflicts

### PR DESCRIPTION
**Pull request by white238**
I attempted to use Serac as a submodule in CMake and ran into some small problems.

### CODE REVIEW

**Comment by white238** [comment]
LGTM

**Review by joshessman-llnl — APPROVED** [review]

File: src/infrastructure/cli.cpp [hunk 1]
```diff
@@ -28,8 +29,13 @@ std::unordered_map<std::string, std::string> defineAndParse(int argc, char* argv
...
     serac::logger::flush();
-    auto err_msg = (e.get_name() == "CallForHelp") ? app.help() : CLI::FailureMessage::simple(&app, e);
-    SLIC_ERROR_ROOT(rank, err_msg);
+    if (e.get_name() == "CallForHelp") {
+      auto msg = app.help();
+      SLIC_INFO_ROOT(rank, msg);
+    } else {
+      auto err_msg = CLI::FailureMessage::simple(&app, e);
+      SLIC_ERROR_ROOT(rank, err_msg);
+    }
```
Comment (joshessman-llnl) [inline src/infrastructure/cli.cpp:38]: Was this needed so the help test has a zero returncode?

Comment (white238) [reply src/infrastructure/cli.cpp:38]: I have run into codes failing when `help` is called.  So while I was touching this code I added a test for it.

I noticed some slightly wrong behavior. `serac --help` should not report an error, it is doing exactly what you asked it to do.

Before this change it looked something like this:

```
[white238]$ bin/serac --help
Logger activated: serac_serial_logger
[ERROR (/usr/WS2/white238/serac/repo/src/infrastructure/cli.cpp:32)]
Serac: a high order nonlinear thermomechanical simulation code
Usage: bin/serac [OPTIONS]

Options:
  -h,--help                   Print this help message and exit
  -i,--input_file TEXT:FILE REQUIRED
                              Input file to use.

** StackTrace of 5 frames **
Frame 1: bin/serac() [0xb62145]
Frame 2: bin/serac() [0x4424e4]
Frame 3: bin/serac() [0x419e05]
Frame 4: __libc_start_main
Frame 5: bin/serac() [0x4196c0]
=====

```

**Comment by white238** [comment]
LGTM

**Review by jamiebramwell — APPROVED** [review]

**Pull request merged**